justfile

```diff
@@ -330,6 +342,57 @@ update:
     cargo +nightly -Z unstable-options update --breaking
     cargo update
 
+# Validate that all required development tools are installed
+validate-tools:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Validating development tools..."
+
+    # Check essential tools
+    missing_tools=()
+
+    if ! command -v jq >/dev/null 2>&1; then
+        missing_tools+=("jq")
+    fi
+
+    if ! command -v file >/dev/null 2>&1; then
+        missing_tools+=("file")
+    fi
+
+    if ! command -v curl >/dev/null 2>&1; then
+        missing_tools+=("curl")
+    fi
+
+    if ! command -v grep >/dev/null 2>&1; then
+        missing_tools+=("grep")
+    fi
+
+    if ! command -v sqlite3 >/dev/null 2>&1; then
+        missing_tools+=("sqlite3")
+    fi
+
+    if ! command -v sqldiff >/dev/null 2>&1; then
+        missing_tools+=("sqldiff")
+    fi
+
+    # Check Linux-specific tools
+    if [[ "$OSTYPE" == "linux"* ]]; then
+        if ! command -v ogrmerge.py >/dev/null 2>&1; then
+            missing_tools+=("ogrmerge.py")
+        fi
+    fi
```

Comment:
For what it's worth, I wanted to be able to run integration tests locally on macOS. I already had `ogrmerge` installed for other work and had no issues. However, because macOS has it's own implementations of `sed` (without `regexp-extended`) and `tr`, I did need to get the `gsed` and `gtr` variants.

I was thinking about putting some work into `test.sh` to automate the handling of this. It might be a little tricky because some macOS users may have installed the GNU commands `--with-default-names`.

Reply:
'As far as I remember, the only reason why I chose the "let regex behave like regex"-option is because I do this always.. Makes this much les confusing.

It should be possible to port to the macos specific version, but I don't currently have access to a macos device.